### PR TITLE
Parse function parameters attributes

### LINKED ISSUE #1397: Support for attributes in function parameters
[#60669](https://github.com/rust-lang/rust/pull/60669) has been merged and I guess the AST needs to be modified to support it.

### PR DESCRIPTION
**Pull request by eupn**
Fixes #1397. The [RFC-2565](https://github.com/rust-lang/rfcs/blob/master/text/2565-formal-function-parameter-attributes.md) specifies `#[attributes]` to function parameters:

```rust
fn foo(#[attr] a, #[unused] b, #[must_use] ...) {
    // ...
}
```

This PR adds those attributes into grammar and to the parser, extending corresponding inline tests.

### CODE REVIEW

File: crates/ra_parser/src/grammar/params.rs [hunk 1]
```diff
@@ -41,9 +44,16 @@ fn list_(p: &mut Parser, flavor: Flavor) {
     let m = p.start();
     p.bump();
     if flavor.type_required() {
+        attributes::outer_attributes(p);
         opt_self_param(p);
     }
     while !p.at(EOF) && !p.at(ket) && !(flavor.type_required() && p.at(T![...])) {
+        attributes::outer_attributes(p);
```
Comment (matklad) [inline crates/ra_parser/src/grammar/params.rs]: ...and let' *one* test  above this line.

All the other attribute test in this PR are better put into a `test_data/parser/ok/parameter_attrs.rs`

File: crates/ra_parser/src/grammar/params.rs [hunk 2]
```diff
@@ -41,9 +44,16 @@ fn list_(p: &mut Parser, flavor: Flavor) {
     let m = p.start();
     p.bump();
     if flavor.type_required() {
+        attributes::outer_attributes(p);
```
Comment (matklad) [inline crates/ra_parser/src/grammar/params.rs]: let's put *one* test just above this line...

File: crates/ra_parser/src/grammar/params.rs [hunk 3]
```diff
@@ -41,9 +44,16 @@ fn list_(p: &mut Parser, flavor: Flavor) {
...
         opt_self_param(p);
     }
     while !p.at(EOF) && !p.at(ket) && !(flavor.type_required() && p.at(T![...])) {
+        attributes::outer_attributes(p);
+
+        if p.at(T![...]) {
```
Comment (matklad) [inline crates/ra_parser/src/grammar/params.rs]: this check is duplicated with the loop header. Let's remove `flavor.type_required() && p.at(T![...])` from the while condition and just breaks out of the loop? Note the `flavor` bit which I think is missing from this `if`

File: crates/ra_parser/src/grammar/params.rs (hunk 1 above)
Comment (eupn) [reply crates/ra_parser/src/grammar/params.rs]: How I should call those tests?

**Comment by matklad** [comment]
bros r+

Thanks!

**Pull request merged**